meilisearch-lib/src/index_controller/dump_actor/mod.rs

```diff
@@ -160,10 +161,31 @@ impl DumpInfo {
 pub fn load_dump(
     dst_path: impl AsRef<Path>,
     src_path: impl AsRef<Path>,
+    ignore_dump_if_db_exists: bool,
+    ignore_missing_dump: bool,
     index_db_size: usize,
     update_db_size: usize,
     indexer_opts: &IndexerOpts,
 ) -> anyhow::Result<()> {
+    let empty_db = crate::empty_db(&dst_path);
+    let src_path_exists = src_path.as_ref().exists();
+
+    if empty_db && src_path_exists {
+        // everything is fine, there is no early return
```

Comment:
can you instead move the relevant code in this branch?

Reply:
Done, I also splitted the code into multiple functions.